src/encode.c

```diff
@@ -721,7 +722,10 @@ PyImaging_LibTiffEncoderNew(PyObject *self, PyObject *args) {
         }
 
         if (!is_core_tag) {
-            PyObject *tag_type = PyDict_GetItem(types, key);
+            PyObject *tag_type;
+            if (PyDict_GetItemRef(types, key, &tag_type) == 0) {
+                PyErr_SetString(PyExc_KeyError, "unknown tag type");
+            }
```

Comment:
This wasn't raising before, so it should probably not raise now either, right?

Also, the `-1` case should be handled appropriately here.